featuremanagement/_featuremanager.py

```diff
@@ -70,14 +73,87 @@ def __init__(self, configuration, **kwargs):
         self._configuration = configuration
         self._cache = {}
         self._copy = configuration.get(FEATURE_MANAGEMENT_KEY)
-
         filters = [TimeWindowFilter(), TargetingFilter()] + kwargs.pop(PROVIDED_FEATURE_FILTERS, [])
 
         for feature_filter in filters:
             if not isinstance(feature_filter, FeatureFilter):
                 raise ValueError("Custom filter must be a subclass of FeatureFilter")
             self._filters[feature_filter.name] = feature_filter
 
+    @staticmethod
+    def _check_default_disabled_variant(feature_flag):
+        if not feature_flag.allocation:
+            return False
+        return FeatureManager._check_variant_override(
+            feature_flag.variants, feature_flag.allocation.default_when_disabled, False
+        )
+
+    @staticmethod
+    def _check_default_enabled_variant(feature_flag):
+        if not feature_flag.allocation:
+            return True
+        return FeatureManager._check_variant_override(
+            feature_flag.variants, feature_flag.allocation.default_when_enabled, True
+        )
+
+    @staticmethod
+    def _check_variant_override(variants, default_variant_name, status):
+        if not variants or not default_variant_name:
+            return status
+        for variant in variants:
+            if variant.name == default_variant_name:
+                if variant.status_override == "Enabled":
+                    return True
+                if variant.status_override == "Disabled":
+                    return False
+        return status
+
+    @staticmethod
+    def _is_targeted(context_id):
+        """Determine if the user is targeted for the given context"""
+        hashed_context_id = hashlib.sha256(context_id.encode()).digest()
+        context_marker = int.from_bytes(hashed_context_id[:4], byteorder="little", signed=False)
+
+        return (context_marker / (2**32 - 1)) * 100
+
+    def _assign_variant(self, feature_flag, **kwargs):
+        if not feature_flag.variants or not feature_flag.allocation:
+            return None
+        if feature_flag.allocation.user:
+            user = kwargs.get("user")
+            if user:
+                for user_allocation in feature_flag.allocation.user:
+                    if user in user_allocation.users:
+                        return user_allocation.variant
+        if feature_flag.allocation.group:
+            groups = kwargs.get("groups")
+            if groups:
+                for group_allocation in feature_flag.allocation.group:
+                    for group in groups:
+                        if group in group_allocation.groups:
+                            return group_allocation.variant
+        if feature_flag.allocation.percentile:
+            user = kwargs.get("user", "")
+            context_id = user + "\n" + feature_flag.name + "\n" + feature_flag.allocation.seed
```

Comment:
What is expected here?
https://github.com/microsoft/FeatureManagement-Dotnet/blob/5c2af2c403c2f1098102a797abb83805379b7652/src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs#L330
https://github.com/microsoft/FeatureManagement-Dotnet/blob/5c2af2c403c2f1098102a797abb83805379b7652/src/Microsoft.FeatureManagement/FeatureManager.cs#L620

In DotNet this value is `user\nhint`, where hint is the seed. but hint is value as the user defined value or "allocation\n{feature_name}", but wouldn't this cause all variants to return the same value if a seed is not provided.

Reply:
I've corrected this, but the change is in the default value in allocaiton.